src/components/ProductDetailInfo.jsx

```diff
@@ -1,19 +1,64 @@
+import { useRef, useContext } from 'react';
 import PropTypes from 'prop-types';
 import Button from 'react-bootstrap/Button';
+import { GlobalContext } from '../context/GlobalContext';
 import ProductTable from './ProductTable';
 
 const ProductDetailInfo = ({ productDetails }) => {
+  const numberInputRef = useRef();
+  const { dispatch, cart } = useContext(GlobalContext);
+  const handleClick = () => {
+    const amountSelected = numberInputRef.current.value;
+    if (!amountSelected) {
+      alert('please add a quantity');
```

Comment:
I think here it would be more useful to have a default of 1 when the product has stock, and disable the button if not selected a quantity